test/unit/spec/LoginRouter_spec.js

```diff
@@ -1046,85 +1046,90 @@ function (Okta, Q, Logger, Errors, BrowserFeatures, WidgetUtil, Bundles, config,
             Expect.deprecated('Use "scopes" instead of "scope"');
           });
       });
-      itp('redirects instead of using an iframe if the responseType is "code"', function () {
+
+      itp('PKCE: redirects, sets the responseMode to "fragment" and sets a code_challenge', function () {
```

Comment:
I like the PKCE here, but can we flag/group/describe/something the other tests to make it clear what kind of flow they are testing so someone (me :) ) in the future doesn't recreate your discovery?

Reply:
The other is the "default" flow for the SIW, I'm not sure what it is called. For some reason I convinced myself that all these tests were not related to anything actually used in production, but were part of early PKCE experiments. I don't think anyone else is going to make that mistake, it sounds pretty crazy. We should review very carefully whenever test conditions are changed.

Reply:
Sorry to be dense, but I think I'm missing something.  All of these changes are within the OIDC test block, so why did they switch to `fragment` instead of `query`?  

Is this OIDC vs AuthN, or was this PKCE vs non-PKCE within OIDC?

Reply:
...that said, my confusion doesn't need to hold up this PR - here you're restoring the way it was before, and within the OIDC block.

Reply:
From my understanding this flow is the standard OIDC "authorization code flow". PKCE is an add-on to this underlying flow, requiring the code challenge. The reasons for the `responseMode` are mostly historic (we are maintaining defaults we have provided previously). A developer could override this in either flow.